### DISCUSSION TITLE
Trigger Azure Data Factory : Not Found

### DISCUSSION THREAD

**Discussion by TamGB** [apache/airflow#21822] [Q&A]
### Apache Airflow Provider(s)

microsoft-azure

### Versions of Apache Airflow Providers

Azure Extension - 3.6.0

### Apache Airflow version

2.2.3

### Operating System

UNIX

### Deployment

Official Apache Airflow Helm Chart

### What happened

I tried following several tutorials on how to trigger a dummy Azure Data Factory pipeline via Airflow, [link 1](https://www.astronomer.io/guides/airflow-azure-data-factory-integration/), [link 2](https://www.astronomer.io/events/recaps/using-airflow-with-azure-data-factory/).

However, regardless of what I do, always obtain the same outcome, the DAG fails, and upon inspecting the Airflow log I get:
```
*** Trying to get logs (last 100 lines) from worker pod airflowadfintegrationdemoextractdatafactorypipelinegetlatestpip ***

*** Unable to fetch logs from worker pod airflowadfintegrationdemoextractdatafactorypipelinegetlatestpip ***
(404)
Reason: Not Found
HTTP response headers: HTTPHeaderDict({'Audit-Id': 'fc84b49e-26c5-4edb-997d-e756469a8df7', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'Date': 'Thu, 24 Feb 2022 16:04:30 GMT', 'Content-Length': '294'})
HTTP response body: b'{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods \\"airflowadfintegrationdemoextractdatafactorypipelinegetlatestpip\\" not found","reason":"NotFound","details":{"name":"airflowadfintegrationdemoextractdatafactorypipelinegetlatestpip","kind":"pods"},"code":404}\n'
```

Which is extremely odd, since the Data Factory does exist, so does the Pipeline, the Subscription ID, Resource Group are all filled in. Has anyone experienced this before?

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by josh-fell** [comment]
Do any other tasks/DAGs complete successfully? The error message from the logs looks like it's an environment or permissions issue to Kube pods (e.g. #11696) rather than something directly related to the Azure provider.

Can you add context to reproduce as well? Anything in the scheduler logs? To help, we'll need a lot more detail to determine if this is a bug.

@potiuk Maybe this gets converted to a discussion for now.

**Comment by TamGB** [comment]
Hi @josh-fell ,

Indeed you are correct, it seems that with other tasks the message is similar, even when they succeed. Could this be a log persistence problem, with Kubernetes deleting the volume associated with the Task once it finishes?

I will have a look in the documentation for anything that can somehow prevent this from happening.
